### PR TITLE
Return proper device name for the coordinator

### PR DESCRIPTION
**Pull request by jloehr**
Use the device path to guess whether it is a serial device and therefore a RaspBee.
The generation can be safely obtained from the firmware version.

### CODE REVIEW

File: zigpy_deconz/zigbee/application.py [hunk 1]
```diff
@@ -281,8 +288,13 @@ def handle_tx_confirm(self, req_id, status):
             )
 
 
-class ConBeeDevice(zigpy.device.Device):
+class DeconzDevice(zigpy.device.Device):
     """Zigpy Device representing Coordinator."""
+    def __init__(self, version: int, device_path: str, *args):
+        super().__init__(*args)
+        is_serial_devcie = re.match(r"/dev/tty(S|AMA)\d+", device_path)
```
Comment (Adminiuga) [inline zigpy_deconz/zigbee/application.py:295]:
```suggestion
        is_serial_device = re.match(r"/dev/tty(S|AMA)\d+", device_path)
```

File: zigpy_deconz/zigbee/application.py [hunk 2]
```diff
@@ -281,8 +288,13 @@ def handle_tx_confirm(self, req_id, status):
...
+        self._model = "RaspBee" if is_serial_devcie else "ConBee"
```
Comment (Adminiuga) [inline zigpy_deconz/zigbee/application.py:296]:
```suggestion
        self._model = "RaspBee" if is_serial_device else "ConBee"
```

File: zigpy_deconz/zigbee/application.py (hunk 1 above)
Comment (Adminiuga) [inline zigpy_deconz/zigbee/application.py:295]: Should we handle `/dev/serial/by-id/` cases?

Comment (jloehr) [reply zigpy_deconz/zigbee/application.py:295]: My current understanding is, that `/dev/serial/by-id/` is used for USB devices. So whatever show up under that is a USB device and therefore a ConBee. Can GPIO shields like the RaspBee also show up under `/dev/serial/by-id/`?

Maybe the name ism't the best, as USB devices are also serial devices, duh. I should probably change that to `is_shield_device` or  `is_gpio_device`.

Comment (Adminiuga) [reply zigpy_deconz/zigbee/application.py:295]: I think you are right.

**Comment by Adminiuga** [comment]
in your dev environment install pre-commit with `pip install pre-commit` then `pre-commit install` so it would run black/isort checks on your commits.

**Pull request merged**